nx/test/nx/defn/grad_test.exs

```diff
@@ -463,6 +463,19 @@ defmodule Nx.Defn.GradTest do
     end
   end
 
+  describe "tan" do
+    defn grad_tan(t), do: grad(t, Nx.tan(t))
+
+    test "computes gradient" do
+      for _ <- 1..100 do
+          multiplier = Nx.random_uniform({}, 0, 10, type: {:u, 32})
+          offset = Nx.random_uniform({}, -1.5, 1.5, type: {:f, 64})
+          t = 3.14159 |> Nx.multiply(multiplier) |> Nx.add(offset)
+          check_grads!(&Nx.tan/1, &grad_tan/1, t)
```

Comment:
The gradient of tan(t) looks like this:
![tan](https://user-images.githubusercontent.com/1389793/108485196-81c4ae00-7251-11eb-99df-7faeaf7cf24e.png)
`check_grads!/4` fails for values close to the asymptotes, so we select `t` to avoid them.

This is why we test tan's gradient separately from the `tuples` above, where `t` is selected by
`t = Nx.random_uniform({}, 0.1, 10.0, type: {:f, 64})`

Reply:
Looks great! Can you please add these notes as a comment to the test above? Thank you!

Reply:
Added!